src/message-fragmentation.md

```diff
@@ -57,6 +67,8 @@ Message IDs should be randomly generated. When retransmitting a message, the mes
 reused. If a new destination is chosen for retransmission however, or if the retransmission is in a
 different session, a new message ID should be generated.
 
+// TODO 128 bit of random is same as a uuid v4 so should prevent collision.
```

Comment:
I think this comes under "justifying design decisions" which I've not tried to do in this spec. FWIW though, 16 bytes is to match the security level of the rest of the system. Note that we only care about collisions with recent message IDs that might still be in eg the fragment assembler; collisions with old message IDs don't matter.